### PR TITLE
Mapped the robot speed to a slider on the joystick.

### PR DESCRIPTION
**Pull request by 10tothe6**
- the robot speed can now be controlled with a slider, it interpolates between the min and max speed constants (like slowmode but not binary)
- some documentation changes
- speed percent now writes to shuffleboard

### CODE REVIEW

**Review by Drake-mcg21 — APPROVED** [review]

**Review by Timatoot — APPROVED** [review]

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -91,7 +91,9 @@ public void teleopInit() {
 
   /** This function is called periodically during operator control. */
   @Override
-  public void teleopPeriodic() {}
+  public void teleopPeriodic() {
+    m_robotContainer.setSpeedPercent();
```
Comment (noah4546) [inline src/main/java/frc/robot/Robot.java:95]: Variables: No under_scores (except constants).

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -97,4 +97,9 @@ public Command getAutonomousCommand() {
     public void eStop() {
         m_robotDrive.setX();
     }
+
+    //sets the speed percentage to use based on the slider on the joystick
```
Comment (noah4546) [inline src/main/java/frc/robot/RobotContainer.java]: Space and Capitalize 'Sets'

**Review by noah4546 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -183,9 +187,14 @@ public void drive(double xSpeed, double ySpeed, double rot, boolean fieldRelativ
...
+    double lerpSpeed = DriveConstants.kMinSpeedMetersPerSecond + (DriveConstants.kMaxSpeedMetersPerSecond
+                     - DriveConstants.kMinSpeedMetersPerSecond) * m_maxSpeed;
+
+    // Convert the commanded speeds into the correct units for the drivetrain,
+    // using the interpolated speed
+    double xSpeedDelivered = xSpeedCommanded * lerpSpeed;
+    double ySpeedDelivered = ySpeedCommanded * lerpSpeed;
     double rotDelivered = m_currentRotation * DriveConstants.kMaxAngularSpeed;
 
     var swerveModuleStates = DriveConstants.kDriveKinematics.toSwerveModuleStates(
```
Comment (noah4546) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: NEVER use var keyword, always use explicit type.

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (10tothe06) [reply src/main/java/frc/robot/Robot.java:95]: I didn't write that variable, Stevon mentioned that in the other PR request. It's part of the REV swerve code.

Comment (Stevon2386) [reply src/main/java/frc/robot/Robot.java:95]: If we are going to change the underscores here we might as well change it everywhere in the code (See [#8](https://github.com/BCHRobotics/Trojans2024/pull/8#discussion_r1454209174))

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 4]
```diff
@@ -57,6 +57,9 @@ public class Drivetrain extends SubsystemBase {
   private double m_currentTranslationDir = 0.0;
   private double m_currentTranslationMag = 0.0;
 
+  // A percentage value (0-1) for the linear speed of the robot
+  public double m_maxSpeed = 0.0;
```
Comment (noah4546) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: Is there a reason this is public?

Do not make any instance or class variable public without a good reason. (Often variables do not need to be explicitly set or gotten – often that happens as a side effect of method calls)

**Review by noah4546 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 3 above)
Comment (noah4546) [reply src/main/java/frc/robot/subsystems/Drivetrain.java]: Will open ticket.

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (noah4546) [reply src/main/java/frc/robot/Robot.java:95]: Yes, will create ticket for this, after I double check the standards.

Comment (noah4546) [reply src/main/java/frc/robot/Robot.java:95]: > I didn't write that variable, Stevon mentioned that in the other PR request. It's part of the REV swerve code.

Do you have a link to the code reference?

Comment (noah4546) [reply src/main/java/frc/robot/Robot.java:95]: Opened issue for this #12

**Review by noah4546 — APPROVED** [review]

**Pull request merged**